### PR TITLE
support podman ps filter regular expressions

### LINKED ISSUE #3394: Support podman ps filter regular expressions like in docker CLI
/kind bug

**Description**

To filter a container on it's exact name, the filter `name=^postgres$` should work, but it doesn't. When using the filter `name=postgres`, containers named `postgres-exporter` and `postgres` will be returned. That's why this filter can be useful.

**Steps to reproduce the issue:**

Podman

1. Run in shell: `podman run --rm -d --name postgres docker://docker.io/library/postgres:11`
2. Run in shell: `podman ps --filter name=^postgres$`

**Describe the results you received:**

No output

**Describe the results you expected:**

```
CONTAINER ID  IMAGE                          COMMAND               CREATED         STATUS             PORTS  NAMES
446492b47e8a  docker.io/library/postgres:11  docker-entrypoint...  31 seconds ago  Up 31 seconds ago         postgres
```

Which docker outputs when podman is replaced with docker in steps to reproduce

**Output of `podman version`:**

```
Version:            1.4.0
RemoteAPI Version:  1
Go Version:         go1.12.5
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.12.5
  podman version: 1.4.0
host:
  BuildahVersion: 1.8.3
  Conmon:
    package: podman-1.4.0-2.fc30.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 0.2.0, commit: 3e5b3b1909019a42c7f5a8d47070f0808bfbb35c'
  Distribution:
    distribution: fedora
    version: "30"
  MemFree: 20706824192
  MemTotal: 33479675904
  OCIRuntime:
    package: containerd.io-1.2.6-3.3.fc30.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc8
      commit: 425e105d5a03fabd737a126ad93d62a9eeede87f
      spec: 1.0.1-dev
  SwapFree: 8589930496
  SwapTotal: 8589930496
  arch: amd64
  cpus: 8
  hostname: fedora-jeffrey
  kernel: 5.1.11-300.fc30.x86_64
  os: linux
  rootless: false
  uptime: 2h 33m 2.19s (Approximately 0.08 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ConfigFile: /etc/containers/storage.conf
  ContainerStore:
    number: 1
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mountopt=nodev,metacopy=on
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  ImageStore:
    number: 13
  RunRoot: /var/run/containers/storage
  VolumePath: /var/lib/containers/storage/volumes
```

**Additional environment details (AWS, VirtualBox, physical, etc.):** 

n/a

### PR DESCRIPTION
**Pull request by QiWang19**
Close #3394 
bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1731107

podman ps --filter use regexp to match the container name.

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
/approve

Thanks for the speedy response here

**Comment by mheon** [comment]
I wonder if we need more than this - we might need to use regexp matching for all filters? Should test against Docker to determine.

**Comment by QiWang19** [comment]
I think no other filters need regular expression after looking through the filters that in common with docker.

File: test/e2e/ps_test.go [hunk 1]
```diff
@@ -332,4 +332,30 @@ var _ = Describe("Podman ps", func() {
...
 		Expect(result.OutputToStringArray()[0]).To(Equal(fullCid))
 	})
+
+	It("podman ps filter name regexp", func() {
+		session := podmanTest.Podman([]string{"run", "-d", "--name", "test1", ALPINE, "top"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		fullCid := session.OutputToString()
+
+		session2 := podmanTest.Podman([]string{"run", "-d", "--name", "test11", ALPINE, "top"})
```
Comment (rhatdan) [inline test/e2e/ps_test.go]: This looks like it should fail, I think this should be named test2.

Comment (rhatdan) [reply test/e2e/ps_test.go]: Never mind, I guess it is test11.

**Pull request merged**